src/Datasource/EntityTrait.php

```diff
@@ -482,7 +482,7 @@ public function setVirtual(array $properties, $merge = false)
             return $this;
         }
 
-        $this->_virtual += $properties;
+        $this->_virtual = array_merge($this->_virtual, $properties);
```

Comment:
It seems it doesn't need it, array_merge() here seems to not duplicate, as the test proves.